airbyte-to-flow/src/interceptors/fix_document_schema.rs

```diff
@@ -0,0 +1,210 @@
+use doc::ptr::Token;
+use serde_json::{value::RawValue, json};
+
+use crate::errors::Error;
+
+
+/// Given a document_schema_json and key ptrs, updates the document_schema to ensure that
+/// the key pointers are always present in the document
+pub fn fix_document_schema_keys(document_schema_json: &RawValue, key_ptrs: Vec<Vec<String>>) -> Result<serde_json::Value, Error> {
+    let mut doc = serde_json::to_value(document_schema_json)?;
+    for key in key_ptrs {
+        let ptr = doc::Pointer::from_vec(&key);
+
+        let mut current = doc::Pointer::empty();
+        for token in ptr.iter() {
+            match token {
+                // Add "minItems" to arrays to ensure the key is always available at that index
+                doc::ptr::Token::Index(idx) => {
```

Comment:
Does this work if an object has index-like property names? Ex https://github.com/estuary/flow/pull/676

Reply:
Ah... no it doesn't actually... I think this is a confusion in JsonPointer's API. I expect `Token::Index` to be an array index, and if there is an index-like property name, I expect to get `Token::Property`

Reply:
I added the test case but at the moment it's failing because the `"0"` property is considered an array index 🤔

Reply:
Good idea, disabled that part and created an issue: https://github.com/estuary/flow/issues/714